File: codechef/BOSSLOSS.java
/* package codechef; // don't place package name! */

import java.math.BigInteger;
import java.util.Scanner;
 
public class Main{
 
    // Driver method
    public static void main(String args[]) throws Exception
    {
        Scanner sc = new Scanner(System.in);
        int t=sc.nextInt();

        
        while(t > 0){
            BigInteger n = sc.nextBigInteger();
            BigInteger m = sc.nextBigInteger();

            BigInteger low=BigInteger.valueOf(1);
            BigInteger ans=BigInteger.valueOf(-1);
            Long l = new Long(10000000000L);
            BigInteger high=BigInteger.valueOf(l);
            if(high.compareTo(n) > 0)
                high=n;
            int r=0;
            
            while(r >= 0){
                BigInteger mid=low.add(high);
                mid=mid.divide(BigInteger.valueOf(2));


                BigInteger x=mid.multiply(mid.add(BigInteger.valueOf(1)));
                x=x.divide(BigInteger.valueOf(2));
                r=x.compareTo(m);
                if(r >= 0){
                    ans=mid;
                    high=mid.subtract(BigInteger.valueOf(1));
                }
                else{
                    low=mid.add(BigInteger.valueOf(1));
                }
                r=high.compareTo(low);

            }

            System.out.println(ans);
            t--;

        }
    }
}

File: codechef/BOSSLOSS.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){

	long long  t,n,m,x=1414213563;
	cin >> t;

	while(t--){
		cin >> n >> m;
		long long  low=1,high,mid,ans=-1,ch;
		high=min(n,x);

		while(low <= high){
			mid=low+(high-low)/2;
			if(mid%2 == 0){
				ch=(mid/2);
				ch*=(mid+1);
			}
			else{
				ch=(mid+1)/2;
				ch*=mid;
			}

			if(ch >= m){
				ans=mid;
				high=mid-1;
			}
			else
				low=mid+1;
		}

		cout << ans << endl;
	}
	return 0;
}